wam_v:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50
  # velocity joint controllers----------------------------------
  right_motor_controller:
    type: velocity_controllers/JointVelocityController
    joint: right_motor_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  left_motor_controller:
    type: velocity_controllers/JointVelocityController
    joint: left_motor_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  # driving force controllers ----------------------------------
  right_driving_force_controller:
    type: driving_force_controller/DrivingForceController
    twist_topic: /wam_v/twist
    motor_command_topic: /wam_v/right_motor_controller/command
    driving_force_command_topic: /wam_v/ship_controller/right_driving_force
    characteristic_curve_file_name: right_motor
    max_rotational_speed: 10
    min_rotational_speed: -10
    gradient_descent:
      tolerance: 0.0001
      alpha: 0.1
    propeller:
      turning_radius: 0.2
      k2: 1
      k1: 1
      k0: 1
      fluid_density: 998.233
  left_driving_force_controller:
    type: driving_force_controller/DrivingForceController
    twist_topic: /wam_v/twist
    motor_command_topic:  /wam_v/left_motor_controller/command
    driving_force_command_topic: /wam_v/ship_controller/left_driving_force
    characteristic_curve_file_name: left_motor
    max_rotational_speed: 10
    min_rotational_speed: -10
    gradient_descent:
      tolerance: 0.0001
      alpha: 0.1
    propeller:
      turning_radius: 0.2
      k2: 1
      k1: 1
      k0: 1
      fluid_density: 998.233
  # ship controllers -------------------------------------------
  #ship_controller:
    #type: ship_controller/ShipController
    #right_motor_command_topic: /wam_v/right_motor_controller/command
    #left_motor_command_topic:  /wam_v/left_motor_controller/command
    #twist_topic: /wam_v/twist
    #right_motor_joint: right_motor_joint
    #left_motor_joint: left_motor_joint
    #linear:
      #max_velocity           : 0.5  # m/s
      #min_velocity           : -0.5 # m/s
      #pid: {p: 100.0, i: 0.01, d: 10.0}
    #angular:
      #max_velocity           : 0.3  # rad/s
      #min_velocity           : -0.3
    #mass: 247
    #izz: 165.39
    #motor_distance: 2.054
